### DISCUSSION TITLE
Use bundler update for lock file maintenance

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#31263] [Suggest an Idea]
### Describe the proposed change(s).

Deleting the lock file and recreating it can result in "platform" changes which are undesirable.

Instead, we should run `bundler update --all`. I'm not _certain_ that it will update everything to the latest possible, including every indirect dependency, but it does seem to preserve platform.

**Comment by RahulGautamSingh** [comment]
Is [this](https://github.com/Rahul-renovate-testing/repro-31117/pull/5/files) an example of an undesriable "platform" change?

**Comment by rarkins** [comment]
I think so. You could also try using the Gemfile/Gemfile.lock from https://github.com/renovatebot/renovate/discussions/30953 to reproduce

**Comment by RahulGautamSingh** [comment]
_Rewriting for clarity_

1. The command `bundler update --all` doesn't work as we want ie. adds undesirable platform changes.

2. Could not find any proper solution for the problem in the rubygems repository discussions/issues.

3. I thought maybe we could extract the platforms from `Gemfile.lock`. And then use the `--add-platform` to resolve deps for extracted platform and also use the `--remove-platform` flag to remove the local platform might work. But I couldn't get the deisred result with this approach.

Not sure how to fix this issue, any help/leads appreciated.

**Reply by rarkins** [reply]
I've converted this to a Discussion until/unless we can confirm this is feasible. If not then we should document the limitation in the Bundler readme.

@andrewpollock could you add any info you have about the bundler update command? @RahulGautamSingh thinks it doesn't behave as you/we hoped it would
